crates/settlement-clients/starknet/src/lib.rs

```diff
@@ -106,44 +117,42 @@ impl SettlementClient for StarknetSettlementClient {
         &self,
         program_output: Vec<[u8; 32]>,
         onchain_data_hash: [u8; 32],
-        onchain_data_size: usize,
+        onchain_data_size: [u128; 2],
     ) -> Result<String> {
         let program_output = slice_slice_u8_to_vec_field(program_output.as_slice());
         let onchain_data_hash = slice_u8_to_field(&onchain_data_hash);
-        let mut calldata: Vec<FieldElement> = Vec::with_capacity(program_output.len() + 2);
-        calldata.extend(program_output);
-        calldata.push(onchain_data_hash);
-        calldata.push(FieldElement::from(onchain_data_size));
-        let invoke_result = self
-            .account
-            .execute(vec![Call {
-                to: self.core_contract_address,
-                selector: *CONTRACT_WRITE_UPDATE_STATE_SELECTOR,
-                calldata,
-            }])
-            .send()
-            .await?;
-        Ok(format!("0x{:x}", invoke_result.transaction_hash))
+        let core_contract: &CoreContract = self.starknet_core_contract_client.as_ref();
+        let onchain_data_size = U256::from_words(onchain_data_size[0], onchain_data_size[1]);
+        let invoke_result = core_contract.update_state(program_output, onchain_data_hash, onchain_data_size).await?;
+
+        Ok(invoke_result.transaction_hash.to_hex_string())
     }
 
     /// Should verify the inclusion of a tx in the settlement layer
     async fn verify_tx_inclusion(&self, tx_hash: &str) -> Result<SettlementVerificationStatus> {
-        let tx_hash = FieldElement::from_hex_be(tx_hash)?;
+        let tx_hash = Felt::from_hex(tx_hash)?;
         let tx_receipt = self.account.provider().get_transaction_receipt(tx_hash).await?;
-        match tx_receipt {
-            MaybePendingTransactionReceipt::Receipt(tx) => match tx.execution_result() {
-                ExecutionResult::Succeeded => Ok(SettlementVerificationStatus::Verified),
-                ExecutionResult::Reverted { reason } => {
-                    Ok(SettlementVerificationStatus::Rejected(format!("Tx {} has been reverted: {}", tx_hash, reason)))
-                }
-            },
-            MaybePendingTransactionReceipt::PendingReceipt(tx) => match tx.execution_result() {
-                ExecutionResult::Succeeded => Ok(SettlementVerificationStatus::Pending),
-                ExecutionResult::Reverted { reason } => Ok(SettlementVerificationStatus::Rejected(format!(
+        let execution_result = tx_receipt.receipt.execution_result();
+        let status = execution_result.status();
+
+        if tx_receipt.block.is_pending() {
+            match status {
+                TransactionExecutionStatus::Succeeded => Ok(SettlementVerificationStatus::Pending),
+                TransactionExecutionStatus::Reverted => Ok(SettlementVerificationStatus::Rejected(format!(
```

Comment:
technically we can do the reverted check before checking for `is_pending` right? will reduce lines of code

Reply:
Updated this